src/Storages/StorageURL.cpp

```diff
@@ -236,6 +241,60 @@ namespace
             };
         }
 
+        // We assume that size of the dst buf isn't less than src_size. Moved from src/Functions/URL/decodeURLComponent.cpp per @bharatnc suggested.
+        static size_t decodeURL(const char * src, size_t src_size, char * dst)
```

Comment:
Do need it at all?  Since we already Poco we can also use [URI::decode](https://docs.pocoproject.org/current/Poco.URI.html#20303) function. See https://github.com/pocoproject/poco/blob/a5b0bb7d96af201e15dea264d7b084ae1db7afcf/Foundation/src/URI.cpp#L669
We don't process billions of URLs here, so what is the reason of reimplementing things avaliable 'for free' in the included libraries?

Reply:
https://github.com/ClickHouse/ClickHouse/pull/33299#discussion_r776518865
This is following @bharatnc 's advice, as you can see above.

Reply:
@bharatnc was for reusing the code from DecodeURLComponent (it's kinda tricky here), I'm for reusing the code from Poco - i.e. URI::decode (which is already there, and is very easy to use). Both of us are against copy & paste & having tons of implementations of the same function in different places.

Reply:
I'm against copy & paste, too. I'm just confused to use decodeURL from decodeURLComponent.cpp as @bharatnc's suggestion or use URI::decode as your suggestion.  That's why I asked there should be an agreement. I will change to URI::decode if that method was adopted in the end.

Reply:
@michael1589 let's use URI::decode from Poco.